packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts

```diff
@@ -157,6 +157,18 @@ export class PrometheusExporter implements MetricExporter {
     });
   }
 
+  /**
+   * Request handler that responds with the current state of metrics
+   * @param request Incoming HTTP request of server instance
+   * @param response HTTP response objet used to response to request
+   */
+  public getMetricsRequestHandler(
```

Comment:
This function names implies you are getting a handler, which is not the what is actually happening. I think it would be smarter to return a middleware that is compatible with express and other web frameworks:

```typescript
public getMetricsRequestHandler() {
  const self = this;
  return function metricsHandler(req, res, next) {
	// req validation?
    // suppress instrumenting this method? optionally?
	// error handling?	

    self.exportMetrics(res);
  };
}
```

Reply:
The idea I was having to return a `raw` handler which has the `IncomingMessage`, and `ServerResponse` arguments as we can't really be sure what the user's framework expects. I was thinking the user could do something like this for Express:

```
app.get('/metrics', (req, res) => {
    exporter.getMetricsRequestHandler(req, res)
})
```

while for Hapi you could do something like:

```
function HapiWrapper(exporter) {
    const handler = exporter.getMetricsRequestHandler()
    return async ( { raw }, h) {
         await handler(raw.req, raw.res)
         return h.close
    } 
}
```

while for Koa something like:

```
const router = app.router()
router.get('/metrics', async (ctx) => {
   await handle(ctx.req, ctx.res)
   ctx.respond = false
})
```

Probably would need to allow the handler to be async but the above the idea to offer most flexibility.  Maybe the name needs to better reflect this?

Reply:
But you have a point maybe it should be returned with `.bind(this)`

Reply:
I agree with @weyert here, its better to use a simpler model that doesn't necessarely match express api so it can be used with every framework (at the cost of having a different usage)

Reply:
Thank you @vmarchaud :)

Reply:
> Does anyone understand why it complains about `sinon` and `@types/sinon`-packages while I have added it to `package.json`?

Because you have not regenerated the package-lock.json files. If you delete the file then bootstrap again it will be fixed.